pkg/proxy/ipvs/proxier.go

```diff
@@ -698,7 +698,7 @@ func CanUseIPVSProxier(ipsetver IPSetVersioner) (bool, error) {
 		err := utilexec.New().Command("modprobe", "--", kmod).Run()
 		if err != nil {
 			glog.Warningf("Failed to load kernel module %v with modprobe. "+
-				"You can ignore this message when kube-proxy is running inside container without mounting /lib/modules", kmod)
+				"You can ignore this message when kube-proxy is running inside container", kmod)
```

Comment:
What does the warning message mean here? Does it mean that I do not need to load those kernel modules? But this PR is using initContainer to load those modules, seems a bit weird to me, can you help elaborate?

Reply:
> What does the warning message mean here? Does it mean that I do not need to load those kernel modules?

ip_vs_* kernel modules is needed. But, it's unnecessary that we should load it via `modprobe`. For example, kube-proxy run inside container and ip_vs_* kernel modules are already installed in host.

> But this PR is using initContainer to load those modules, seems a bit weird to me, can you help elaborate?

If kube-proxy is always installed by the scripts/manifests which this PR has covered, we can change the warning to error, but actually it's ok.

Reply:
Thanks for replaying, we have this warning when `modprobe` command return err, if we use init container then we don't need load kernel modules inside kube-proxy container, so can ignore this message. The follow code will check if kernel loaded.

Reply:
Thanks @Lion-Wei and @m1093782566 , still have some question: If the `modprobe` return error, then it should means that some modules are not loaded, I think that we should return error for such case, as the ipvs may not work well if some modules are not loaded even running in container?

Reply:
@gyliu513 

> If the modprobe return error, then it should means that some modules are not loaded

NO.

Suppose you run kube-proxy in a container. When you execute `modprobe ip_vs` command, you will get an error since there is no kernel module in container by default. However, you will find some kernel modules in `/proc/modules`.

That's why we do 2 steps:

1 modprobe

2. "cut", "-f1", "-d", " ", "/proc/modules"

We should consider both container and non-container cases.

Reply:
thanks @m1093782566 

So if the host has kernel modules in `/proc/modules`, then we do not need to load those modules in container any more if the container is using security mode?

Reply:
Got it, as the initContainer already done the work, how about add more info claim that the `initContainer already helped to do the modprobe work`?

Reply:
no? kube-proxy process should not be aware of init container? In other words, process should not care how it's deployed.

Reply:
so, why ignore this message when proxy is running in container? How to handle the case if there is no init container and kube proxy do not mount /lib/modules either?

Reply:
> so, why ignore this message when proxy is running in container?

Because users can deploy kube-proxy in anyway they like, they may don't use init container.

> How to handle the case if there is no init container and kube proxy do not mount /lib/modules either?

If kube-proxy is running in container and didn't mount /lib/modules, it still has a chance to run IPVS mode.